scene/gui/code_edit.cpp

```diff
@@ -101,6 +103,71 @@ void CodeEdit::_notification(int p_what) {
 				}
 			}
 
+			/* Codeblock Guidelines */
+			if (codeblock_guidelines_enabled) {
+				const int xmargin_beg = style_normal->get_margin(SIDE_LEFT) + get_total_gutter_width();
+				const float space_width = font->get_char_size(' ', font_size).width;
+				const double v_scroll = get_v_scroll();
+				const double h_scroll = get_h_scroll();
+				Vector<int> block_ends;
+				for (int i = 0; i < get_line_count(); i++) {
```

Comment:
We should only draw what is on screen, rather then the entire text.

Reply:
So this would require caching block start and endings whenever the text is changed and starting from there. The change may take a while to implement, but it has potential to make the additional "extra highlight" of the code block the caret is in more reliable.

Reply:
We can certainly calculate from line 0 to end of the visible viewport, so long as we don't make any unnecessary calls to the  `RenderingServer`. `TextEdit` already does quite a lot around this. Will probably require some form of benchmarking to see if it's worth caching.

Reply:
So what you're suggesting *for now* is to **at least** make sure that the RenderingServer doesn't draw guidelines that cannot be seen?

Reply:
Yeah whilst stopping calculation at the end of the visible viewport rather then all the way to the end of the script. I think caching will be faster, but better to benchmark first to make sure the extra complexity is worth it.